dotnetv3/EventBridge Scheduler/Scenarios/SchedulerWorkflow.cs

```diff
@@ -0,0 +1,655 @@
+                    .AddTransient<SchedulerWrapper>()
+            )
+            .Build();
+
+        if (_interactive)
+        {
+            _logger = LoggerFactory.Create(builder => { builder.AddConsole(); })
+                .CreateLogger<SchedulerWorkflow>();
+
+            _schedulerWrapper = host.Services.GetRequiredService<SchedulerWrapper>();
+            _amazonCloudFormation = host.Services.GetRequiredService<IAmazonCloudFormation>();
+        }
+
+        Console.WriteLine(new string('-', 80));
+        Console.WriteLine("Welcome to the Amazon EventBridge Scheduler Workflow.");
+        Console.WriteLine(new string('-', 80));
+
+        try
+        {
+            Console.WriteLine(new string('-', 80));
+            var prepareSuccess = await PrepareApplication();
+            Console.WriteLine(new string('-', 80));
+
+            if (prepareSuccess)
+            {
+                Console.WriteLine(new string('-', 80));
+                await CreateOneTimeSchedule();
+                Console.WriteLine(new string('-', 80));
+
+                Console.WriteLine(new string('-', 80));
+                await CreateRecurringSchedule();
+                Console.WriteLine(new string('-', 80));
+            }
+
+            Console.WriteLine(new string('-', 80));
+            await Cleanup();
+            Console.WriteLine(new string('-', 80));
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "There was a problem with the workflow, initiating cleanup...");
+            _interactive = false;
+            await Cleanup();
+        }
+
+        Console.WriteLine("Amazon EventBridge Scheduler workflow completed.");
+    }
+
+    /// <summary>
+    /// Prepares the application by creating the necessary resources.
+    /// </summary>
+    /// <returns>True if the application was prepared successfully.</returns>
+    public static async Task<bool> PrepareApplication()
+    {
+        Console.WriteLine("Preparing the application...");
+        try
+        {
+            // Prompt the user for an email address to use for the subscription.
+            Console.WriteLine("\nThis example creates resources in a CloudFormation stack, including an SNS topic" +
+                          "\nthat will be subscribed to the EventBridge Scheduler events. " +
+                          "\n\nYou will need to confirm the subscription in order to receive event emails. ");
+
+            var emailAddress = PromptUserForEmail();
+
+            // Prompt the user for a name for the CloudFormation stack
+            _stackName = PromptUserForStackName();
+
+            // Deploy the CloudFormation stack
+            var deploySuccess = await DeployCloudFormationStack(_stackName, emailAddress);
+
+            if (deploySuccess)
+            {
+                // Create a schedule group for all workflow schedules
+                await _schedulerWrapper.CreateScheduleGroupAsync(_scheduleGroupName);
```

Comment:
This wrapper returns a bool to indicate success or failure, but you're ignoring it here.

Reply:
I updated to return it to the parent, but at the highest level if it fails, the user will be notified, but the rest of the workflow can continue.